### PR TITLE
(GH-2016) Getting Started with Scripts Docs(DO NOT MERGE)

### PR DESCRIPTION
**Pull request by adreyer**
This is an example of what a good on-ramp to running scripts with bolt
and sharing those scripts with wrapper plans would look like. It
requires some new features such as powershell params and yaml env_vars.

### CODE REVIEW

File: documentation/scripts.md [hunk 1]
```diff
@@ -0,0 +1,402 @@
...
+## When to use scripts vs tasks
+
+There are a lot of similarities between bolt's ability to run script plans like
+the one above and it's ability to run tasks. In general scripts are easer to
+write and debug and should be favored unless some feature of the Task API is
+needed. It's always easy to convert a script into a task especially if it
+accepts input through environment variables. Unless you need one of the
+following features write a script.
+
+* Typed input: Scripts can only accept strings so if you want to pass strucutred objects you may need turn it into a task.
```
Comment (adreyer) [inline documentation/scripts.md:399]: add noop as a reason to write a task.

File: documentation/scripts.md [hunk 2]
```diff
@@ -0,0 +1,402 @@
...
+
+```
+param(
+  [Parameter(Mandatory=$true)]
+  [String]$Path,
+  [Switch]$Name
+)
+
+
+Get-ChildItem "-Path:$Path" "-Name:$Name"
```
Comment (jpogran) [inline documentation/scripts.md:321]:
```suggestion
Get-ChildItem -Path $Path -Name $Name
```

File: documentation/scripts.md [hunk 3]
```diff
@@ -0,0 +1,402 @@
...
+```
+
+You can now run this script with
+
+```
+bolt script run files/list-files.ps -t target1 "\-Path:c:\\"
```
Comment (jpogran) [inline documentation/scripts.md:327]:
```suggestion
bolt script run files/list-files.ps1 -t target1 "\-Path c:\\"
```

**Comment by lucywyman** [comment]
Superseded by #3016

**Pull request closed without merging**